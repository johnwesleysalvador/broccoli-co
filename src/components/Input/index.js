import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormControl from '@material-ui/core/FormControl';
import styled from 'styled-components';

const Wrapper = styled(FormControl)`
    && {
        margin: 15px 0;
        position: relative;
    }
`;

const Error = styled(FormHelperText)`
    position: absolute;
    top: 100%;
`;

const Input = ({ input: { ...input }, meta: { touched, error }, ...rest }) => (
    <Wrapper>
        <TextField {...rest} {...input} {...error && touched && { error: true }} />
        {error &&
            touched && (
            <Error error id="name-helper-text">
                {error}
            </Error>
        )}
    </Wrapper>
);

Input.propTypes = {
    input: PropTypes.shape({
        value: PropTypes.string,
        name: PropTypes.string.isRequired,
        onChange: PropTypes.func.isRequired,
    }),
    meta: PropTypes.shape({
        touched: PropTypes.bool.isRequired,
        error: PropTypes.string,
    }),
    label: PropTypes.string.isRequired,
};

Input.defaultProps = {
    input: { name: '', onChange: () => {} },
    meta: { touched: false },
};

export default Input;
