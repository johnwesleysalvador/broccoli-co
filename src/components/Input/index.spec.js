import React from 'react';
import renderer from 'react-test-renderer';

import Input from '.';

describe('<Input />', () => {
    test('Should render Input', () => {
        const render = renderer
            .create(<Input name="name" label="Full name" touched={false} />)
            .toJSON();
        expect(render).toMatchSnapshot();
    });
});
