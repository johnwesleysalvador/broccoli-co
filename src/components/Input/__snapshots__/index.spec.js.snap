// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should render Input 1`] = `
<div
  className="MuiFormControl-root-1 sc-bdVaJa goOnnM"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-1"
    onBlur={[Function]}
    onFocus={[Function]}
    touched={false}
  >
    <label
      className="MuiFormLabel-root-10 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9"
      data-shrink={false}
      htmlFor={undefined}
    >
      Full name
    </label>
    <div
      className="MuiInput-root-15 MuiInput-formControl-16 MuiInput-underline-19"
    >
      <input
        aria-invalid={false}
        aria-required={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-23"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name=""
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="text"
        value={undefined}
      />
    </div>
  </div>
</div>
`;
